client/src/app/layout/HeaderApp/SSOMenu.tsx

```diff
@@ -30,19 +32,21 @@ export const SSOMenu: React.FC = () => {
         md: "visible",
       }} /** this user dropdown is hidden on mobile sizes */
     >
-      {keycloak && (
-        <Dropdown
-          isPlain
-          position="right"
-          onSelect={onDropdownSelect}
-          isOpen={isDropdownOpen}
-          toggle={
-            <DropdownToggle id="sso-actions-toggle" onToggle={onDropdownToggle}>
-              {(keycloak.idTokenParsed as any)["preferred_username"]}
-            </DropdownToggle>
-          }
-          dropdownItems={[
-            <DropdownGroup key="sso">
+      <Dropdown
+        isPlain
+        position="right"
+        onSelect={onDropdownSelect}
+        isOpen={isDropdownOpen}
+        toggle={
+          <DropdownToggle id="sso-actions-toggle" onToggle={onDropdownToggle}>
+            {keycloak
+              ? (keycloak.idTokenParsed as any)["preferred_username"]
+              : "Username"}
+          </DropdownToggle>
+        }
+        dropdownItems={[
+          <DropdownGroup key="sso">
+            {keycloak && (
```

Comment:
(To followup above comment)
And here use `{keycloak?.isAuthenticated && (...`
or {`isAuthRequired && keycloak?.isAuthenticated && (...`